src/main/java/uk/gov/digital/ho/hocs/audit/repository/AuditRepository.java

```diff
@@ -1,24 +1,40 @@
 package uk.gov.digital.ho.hocs.audit.repository;
 
 import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.jpa.repository.QueryHints;
 import org.springframework.data.repository.CrudRepository;
 import org.springframework.stereotype.Repository;
 import uk.gov.digital.ho.hocs.audit.repository.entity.AuditEvent;
 
+import javax.persistence.QueryHint;
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.util.List;
 import java.util.UUID;
 import java.util.stream.Stream;
 
+import static org.hibernate.annotations.QueryHints.READ_ONLY;
+import static org.hibernate.jpa.QueryHints.HINT_CACHEABLE;
+import static org.hibernate.jpa.QueryHints.HINT_FETCH_SIZE;
+
 @Repository
 public interface AuditRepository extends CrudRepository<AuditEvent, String>, AuditRepositoryCustom, AuditRepositoryLatestEvents {
 
+    @QueryHints(value = {
+            @QueryHint(name = HINT_FETCH_SIZE, value = "1"),
```

Comment:
Is 1 a real valid size for this?

Reply:
https://www.geekyhacker.com/2019/03/26/high-performance-data-fetching-using-spring-data-jpa-stream/
this is for mysql where the magic number is `Integer.MIN_VALUE`, the magic number for postgres is '1'